lahja/asyncio/endpoint.py

```diff
@@ -368,16 +390,77 @@ def run(self, *args, **kwargs):  # type: ignore
     async def start(self) -> None:
         if self.is_running:
             raise RuntimeError(f"Endpoint {self.name} is already running")
+
         self._receiving_loop_running = asyncio.Event()
         self._receiving_queue = asyncio.Queue()
 
+        self._subscription_updates_running = asyncio.Event()
+        self._subscription_updates_condition = asyncio.Condition()
+        self._subscription_updates_queue = asyncio.Queue()
```

Comment:
These could be initialized in `__init__` which would save you from needing to declare them at the beginning of this class and would also make it easier to reason about the potential for `None` dereferences.

The `await *.running()` pattern sounds like it could be pulled out into a helper which removes the potentially `None` instance attribute entirely. Something like `await self.start_coro(self._process_subscription_updates)` which doesn't return until the coro has started.

Reply:
I'm going to leave this for now and open an issue to track as I've seen this issue arising as well but I don't want to conflate this PR with an extra fix that I'm not sure I know the right solution for.

Reply:
https://github.com/ethereum/lahja/issues/106